mm2src/coins/nft/nft_structs.rs

```diff
@@ -169,9 +173,16 @@ pub struct WithdrawErc721 {
     pub(crate) fee: Option<WithdrawFee>,
 }
 
+#[derive(Clone, Deserialize)]
+pub struct WithdrawNftReq {
+    pub(crate) url: String,
```

Comment:
It is needed to [withdraw erc1155 token type](https://github.com/KomodoPlatform/atomicDEX-API/blob/nft-reverse-proxy/mm2src/coins/nft.rs#L207) in current implementation.
As amount of erc1155 NFT could be > 1, we need to check that owner has enough "NFT" balance. So I created [find_wallet_amount](https://github.com/KomodoPlatform/atomicDEX-API/blob/nft-reverse-proxy/mm2src/coins/nft.rs#L266) function to access info about all nfts owned by the user.

ps: I cant use [get_nft_metadata](https://github.com/KomodoPlatform/atomicDEX-API/blob/nft-reverse-proxy/mm2src/coins/nft.rs#L90) function, bcz it returns info about the latest owner. ERC1155 can have more than 1 owners, it is fine to call this finction to get the info about `name`, `token_uri`, `metadata` etc since they are common to erc1155 tokens with identical token_id, but owner in response could be different, so amount could be different.

Reply:
I just thought that hard coding is a bad practice and providing url in request enable people to use their own proxy if they want. But if you ok with hard coded endpoint as it was I can return it.

Reply:
Yeah, this seems reasonable. But as far as I see this is send for each withdraw request, unlike other urls

Reply:
btw I don't have hard feeling on this - we can ship this with it. It just didn't look quite right to me(sending url with each request)

Reply:
> It just didn't look quite right to me(sending url with each request)
No worry when db support will be added only `update_nft` and `refresh_metadata` API methods will use url, bcz only these methods will be sending requests to moralis to get new info.
GUI told that its fine for them to change requests a little bit I will just need to ping them and provide up to date info about req and res forms.

Reply:
> But as far as I see this is send for each withdraw request, unlike other urls

Well yeah I ask url for all erc types in withdraw method, but use it only for "type":"withdraw_erc1155", I just didnt want to add new param in withdraw_erc1155 structure. Once db will be added I just remove url, other fields will be the same.

Reply:
Usually RPC urls should be included in the enable calls and saved in the coin's fields as part of a client struct, but NFT is a special case since we don't have an enable call for them and defining a coin configuration to an NFT or NFTs in general makes no sense. I still think defining an enable call for NFTs, and some kind of context/coin or client to save the url/s on enable and maybe other important configs in the future if needed is the optimal solution to this url problem. This enable call can be even part of `enable_eth_with_tokens` if needed.

@laruh please consider this in future PRs if you think it will make things easier.

Reply:
@shamardy nice hint, thanks.
personally I dont see using urls in couple future NFT methods as a problem, as NFT is a special case right now.
But yeah I think we cant avoid making some nft configs or adding some enable actions in NFT swaps implementation (related to this [thread](https://github.com/KomodoPlatform/atomicDEX-API/pull/1704#discussion_r1136114811) in previous pr).
I will safe this hint for swaps implementation.